src/web/handlers.rs

```diff
@@ -212,3 +195,46 @@ fn fallback_response(settings: &Settings, tiles: &cache::Tiles) -> HttpResponse
         tiles.to_response(settings.cache_control_header)
     }
 }
+
+/// Check if the tile request should be responded early.
+///
+/// This allows us to short circuit requests if:
+///   - they are not sent from regions of the live markets
+///   - they are sent from unknown device types (`form_factor == "other"`)
+///
+/// It returns a proper response if the early response is desired.
+/// Otherwise, it returns None.
+async fn maybe_early_respond(
+    settings: &Settings,
```

Comment:
suggestion (non-blocking): Can you use `state.settings` to get at the settings, instead of passing it as a separate argument?